transformers/avayaVoiceChannel/transformer.ts

```diff
@@ -0,0 +1,231 @@
+        }
+		return output;
+	},
+
+	/**
+	 * This transformer is executed when the Flow execution has finished.
+	 * For REST based transformers, the final output will be sent to
+	 * the user.
+	 *
+	 * @param processedOutput This is the final object that will be sent to the user.
+	 * It is therefore structured according to the Endpoint channel of the transformer.
+	 *
+	 * @param outputs This is an array of all of the outputs that were output by the Flow.
+	 * These will be merged to create the 'processedOutput' object.
+	 * 
+	 * @param userId The unique ID of the user.
+	 * @param sessionId The unique ID for this session. Can be used together with the userId
+	 * to retrieve the sessionStorage object.
+	 * @param endpoint The configuration object for the used Endpoint.
+	 * @param response The express response object that can be used to send a custom response back to the user.
+	 *
+	 * @returns An object that will be sent to the user, unchanged. It therefore has to have the
+	 * correct format according to the documentation of the specific Endpoint channel.
+	 */
+	handleExecutionFinished: async ({ processedOutput, outputs, userId, sessionId, endpoint, response }) => {
+		console.log("handleExecutionFinished called");
+		let url = COGNIGY_BASE_URL + endpoint.URLToken;
+		const sessionStorage = await getSessionStorage(userId, sessionId);
+		let r = 'default';
+		switch (sessionStorage.cpaas_channel) {
+			case 'call':
+				r = getCPaaSCallCmd(sessionStorage, url, outputs);
+				break;
+			case 'sms':
+				r = getCPaaSSmsCmd(sessionStorage, url, outputs);
+				break;
+			default:
+				break;
+		}
+		response.send(r);
+		console.log('CPaaS cmd='.concat(r));
+		return processedOutput;
+	}
+});
+
+const getCPaaSCallCmd = (sessionStorage, url, outputs) => {
+	let dial = sessionStorage.dial;
+	let hangup = sessionStorage.hangup;
+	let dtmf = sessionStorage.dtmf;
+	let record = sessionStorage.record ? sessionStorage.record : '';
+	let conference = sessionStorage.conference;
+	let ctrlcmd = dial | hangup | conference;
+	sessionStorage.dial = false;
+	sessionStorage.hangup = false;
+	sessionStorage.dtmf = false;
+	sessionStorage.record = null;
+	sessionStorage.conference = false;
+	let prompt = outputs.map((t) => {return t.text}).join('\n');
+	let numDigits = dtmf ? 'numDigits="1"' : '';
+	let gather = '<Gather method="POST" action="'+url+'" input="speech dtmf" language="en-US" timeout="3" '+numDigits+'>'+prompt+'</Gather>';
+	let r = '<Response>' + (record) + (ctrlcmd ? prompt : gather) + '</Response>';
+	return (r);
+};
+
+const getCPaaSSmsCmd = (sessionStorage, url, outputs) => {
+	let text = outputs.map((t) => {return t.text}).join('\n');
+	let From = sessionStorage.To;
+	let To = sessionStorage.From;
+	let FromTo = ' From="'+From+'" To="'+To+'"';
+	let r = '<Response><Sms action="'+url+FromTo+'>'+text+'</Sms></Response>';
+	return (r);
+}
+
+
+
+interface CPaaSRequest {
+	body: CPaaSBody;
+}
+
+interface CPaaSBody {
```

Comment:
Please move the interface to the top of the file.

Reply:
made the changes per suggestion.